### DISCUSSION TITLE
Is there a way to convert a regular model.pt to a CTranslate2 model?

### DISCUSSION THREAD

**Discussion by Muertoe** [SYSTRAN/faster-whisper#235] [Q&A] [answered]
I realize there may be an obvious answer for this but I've been looking for many hours and couldn't find any. It may not even be possible.

My company's data scientist is no longer with the company and he was using OpenAI Whisper for the ASR. All we have is the model.pt which he loads like this and works totally fine albeit slow:

`whisper.load_model("whisper/pretrained/model.pt", device="cuda", in_memory=True)`

I would like to be able to use this same model in faster-whisper and I realize I need to convert it. Is there a way to convert this model to the necessary model if that's all I have to work with?

Thank you very much for any help.

**Comment by guillaumekln** [comment, marked as answer]
Is it a fine-tuned model or one of the pretrained model from OpenAI?

If it is one of the pretrained model from OpenAI, you should find what model size it corresponds to (e.g. "small", "medium", etc.) and then use the same size in faster-whisper as shown in the README (no need for a conversion).

If it is a fine-tuned model, you should first [convert it to a Hugging Face model](https://github.com/huggingface/transformers/blob/main/src/transformers/models/whisper/convert_openai_to_hf.py) and then use the [conversion script described in the README](https://github.com/guillaumekln/faster-whisper#model-conversion).

**Reply by Muertoe** [reply]
Thank you, that worked.

**Reply by Muertoe** [reply]
One more question. The first part seemed to work (convert to HF model), but now I'm faced with this when using the conversion script:
OSError: Can't load tokenizer for './newmodel'. If you were trying to load it from 'https://huggingface.co/models', make sure you don't have a local directory with the same name. Otherwise, make sure './newmodel' is the correct path to a directory containing all relevant files for a WhisperTokenizer tokenizer.

This is what was generated by the first step:
![image](https://github.com/guillaumekln/faster-whisper/assets/133402150/907078c1-e6d2-4f33-a9ac-dff810592be3)

**Reply by guillaumekln** [reply]
See #220. You should add some tokenizer files from the base model.

**Reply by Muertoe** [reply]
Thank you again! That did it.

**Comment by Muertoe** [comment]
It's a fine-tuned model.

I'll try out that link. Thank you very much.

**Comment by shahzebali42** [comment]
@guillaumekln I have a finetuned huggingface whisper model saved locally, I need to convert it ct2 to run with faster-whisper. I have used the script in the readme but it is returning "zsh: segmentation fault  ct2-transformers-converter --model  --output_dir whisper-urdu-v2-ct2    " . I had checked other issues as well, tried refreshing env and creating new env etc. but no luck, getting the same error. Please help me out.

**Reply by guillaumekln** [reply]
Please do not post your issue in multiple places (https://github.com/guillaumekln/faster-whisper/issues/137#issuecomment-1621581645, https://github.com/guillaumekln/faster-whisper/issues/18#issuecomment-1621608671). It just makes it more difficult for anyone to help you.

As far as I understand you running into this issue on Mac, but I don't have a Mac to reproduce the issue. Hopefully someone else will help you.

**Reply by amiroshn** [reply]
Having the same issue. Any Idea how to fix it?
